### PR TITLE
Use JsonTypeDef for GetStateValidators request

### PR DESCRIPTION
**Pull request by courtneyeh**
#7762 

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

**Comment by courtneyeh** [comment]
Note: The new type def request is now used in `RemoteValidatorApiHandler`, but the old call is still used in `VoluntaryExitCommand`. I have been working on this, but can be done in a seperate PR as seems more complex.

File: ethereum/json-types/src/main/java/tech/pegasys/teku/ethereum/json/types/beacon/StateValidatorData.java [hunk 1]
```diff
@@ -13,50 +13,16 @@
...
-
-  public static final SerializableTypeDefinition<ObjectAndMetaData<List<StateValidatorData>>>
-      STATE_VALIDATORS_RESPONSE_TYPE =
-          SerializableTypeDefinition.<ObjectAndMetaData<List<StateValidatorData>>>object()
-              .name("GetStateValidatorsResponse")
-              .withField(
-                  EXECUTION_OPTIMISTIC, BOOLEAN_TYPE, ObjectAndMetaData::isExecutionOptimistic)
-              .withField(FINALIZED, BOOLEAN_TYPE, ObjectAndMetaData::isFinalized)
-              .withField("data", listOf(STATE_VALIDATOR_DATA_TYPE), ObjectAndMetaData::getData)
-              .build();
```
Comment (rolfyone) [inline ethereum/json-types/src/main/java/tech/pegasys/teku/ethereum/json/types/beacon/StateValidatorData.java:58]: this looks fairly clean, why did we need to change to a builder?

Comment (courtneyeh) [reply ethereum/json-types/src/main/java/tech/pegasys/teku/ethereum/json/types/beacon/StateValidatorData.java:58]: To be able to create a `DeserializableTypeDefinition`

**Review by rolfyone — APPROVED** [review]
LGTM

**Pull request merged**